posts/google-summer-of-code-fellows/gsoc-2024-fellows.qmd

```diff
@@ -0,0 +1,294 @@
+
+It mainly covers the three cases as mentioned in the R-research paper:
+
+1. Switch Case:
+
+*Condition*: If the gap between the two treatment events is smaller than the combinationWindow, but the gap is not equal to the duration of either event.
+*Action*: The event_end_date of the previous treatment is set to the event_start_date of the current treatment. This effectively "shifts" the previous treatment’s end date to eliminate the gap, merging the treatments into one continuous period.
+*Purpose*: This ensures that treatment gaps that are too small (less than combinationWindow) are treated as part of the same treatment window.
+
+```julia
+if -gap_previous < combinationWindow && !(-gap_previous in [duration_era, prev_duration_era])
+    treatment_history[i-1, :event_end_date] = treatment_history[i, :event_start_date]
+```
+Here is the pictorial representation for the same:
+![](./case1.png)
+
+2. FRFS (First Row, First Shortened):
+
+*Condition*: If the gap is larger than or equal to the combinationWindow, or the gap equals the duration of one of the two treatments, and the first treatment ends before or on the same date as the second treatment.
+*Action*: A new row is created where the second treatment’s event_end_date is set to the end date of the first treatment. This preserves the overlap but ensures that the earlier treatment period stays intact.
+*Purpose*: This prevents unnecessary truncation of the first treatment if it spans the entire overlap window.
+
+```julia
+elseif -gap_previous >= combinationWindow || -gap_previous in [duration_era, prev_duration_era]
+    if treatment_history[i-1, :event_end_date] <= treatment_history[i, :event_end_date]
+        new_row = deepcopy(treatment_history[i, :])
+        new_row.event_end_date = treatment_history[i-1, :event_end_date]
+        append!(treatment_history, DataFrame(new_row'))
+```
+Here is the pictorial representation for the same:
+![](./case2.png)
+
+3. LRFS (Last Row, First Shortened):
+
+*Condition*: If the gap is larger than or equal to the combinationWindow, or the gap equals the duration of one of the treatments, and the first treatment ends after the second treatment.
+*Action*: The current treatment’s event_end_date is adjusted to match the event_end_date of the previous treatment.
+*Purpose*: This handles cases where the second treatment's window should be shortened to prevent overlap with the previous treatment, merging them into a single continuous window.
+
+```julia
+else
+    treatment_history[i, :event_end_date] = treatment_history[i-1, :event_end_date]
+```
+Here is the pictorial representation for the same:
+![](./case3.png)
+
+
+> *Note:* However, There are a few things left to cover here, most of which is the documentations and writing the test-suite for the same.
+
+Here's a refined version of what you wrote:
+
+---
+
+# Contributions Beyond Coding
```

Comment:
Oooh it would be great if you could also include those DocumenterVitepress PRs you made too as well as the ones to OMOPCDMCohortCreator.jl -- you have a great story to tell!